src/custom-elements.js

```diff
@@ -646,37 +672,42 @@ let Deferred;
   CustomElementRegistry.prototype['_observeRoot'] = CustomElementRegistry.prototype._observeRoot;
   CustomElementRegistry.prototype['_addImport'] = CustomElementRegistry.prototype._addImport;
 
-  // patch window.HTMLElement
+  patchElement('HTMLElement')
+  var htmlElementSubclasses = [ 'Button', 'Canvas', 'Data', 'Head', 'Mod', 'TableCell', 'TableCol', 'Anchor', 'Area', 'Base', 'Body', 'BR', 'DataList', 'Details', 'Dialog', 'Div', 'DList', 'Embed', 'FieldSet', 'Form', 'Heading', 'HR', 'Html', 'IFrame', 'Image', 'Input', 'Keygen', 'Label', 'Legend', 'LI', 'Link', 'Map', 'Media', 'Menu', 'MenuItem', 'Meta', 'Meter', 'Object', 'OList', 'OptGroup', 'Option', 'Output', 'Paragraph', 'Param', 'Picture', 'Pre', 'Progress', 'Quote', 'Script', 'Select', 'Slot', 'Source', 'Span', 'Style', 'TableCaption', 'Table', 'TableRow', 'TableSection', 'Template', 'TextArea', 'Time', 'Title', 'Track', 'UList', 'Unknown'];
+  for (let index in htmlElementSubclasses) {
+      patchElement(`HTML${htmlElementSubclasses[Number(index)]}Element`);
+  }
 
-  /** @const */
-  const origHTMLElement = window.HTMLElement;
-  CustomElementRegistry.prototype['nativeHTMLElement'] = origHTMLElement;
-  /**
-   * @type {function(new: HTMLElement)}
-   */
-  const newHTMLElement = function HTMLElement() {
-    const customElements = _customElements();
+  function patchElement(varName) {
```

Comment:
This function is a generic function that is called with HTMLElement, HTMLButtonElement, etc. The code is almost exactly the same as it was before, except generic. All differences have been called out in review comments below.